repo/content/content_manager_indexes.go

```diff
@@ -147,3 +166,55 @@ func (bm *Manager) addIndexBlobsToBuilder(ctx context.Context, bld packIndexBuil
 
 	return nil
 }
+
+func addBlobsToIndex(ndx map[blob.ID]*IndexBlobInfo, blobs []blob.Metadata) {
+	for _, it := range blobs {
+		if ndx[it.BlobID] == nil {
+			ndx[it.BlobID] = &IndexBlobInfo{
+				Metadata: blob.Metadata{
+					BlobID:    it.BlobID,
+					Length:    it.Length,
+					Timestamp: it.Timestamp,
+				},
+			}
+		}
+	}
+}
+
+func removeCompactedIndexes(ctx context.Context, m map[blob.ID]*IndexBlobInfo, compactionLogs map[blob.ID]*compactionLogEntry, markAsSuperseded bool) {
+	var validCompactionLogs []*compactionLogEntry
+
+	for _, cl := range compactionLogs {
+		// only process compaction logs for which we have found all the outputs.
+		haveAllOutputs := true
```

Comment:
This is just for safety right? When can it happen that an output index blob was not found after the corresponding compaction blob was written.

Reply:
due to eventual consistency it's possible that:

- none will be present
- log will be present, but not index file
- index will be present but not the log
- both will be present